src/visitors/semantic_analysis_visitor.cpp

```diff
@@ -22,6 +23,28 @@ bool SemanticAnalysisVisitor::check(const ast::Program& node) {
     }
     /// -->
 
+    /// <-- This code is for check 3
```

Comment:
check 4 I guess?

Reply:
Ah yes! I fixed the merge conflicts but didn't realize the above needs to be changed! thanks!